stores/sql/sqlite/migrations/main/migration_00018_archived_contracts.sql

```diff
@@ -0,0 +1,135 @@
+
+    `contract_price` text,
+    `initial_renter_funds` text,
+
+    `delete_spending` text,
+    `sector_roots_spending` text,
+    `fund_account_spending` text,
+    `upload_spending` text
+);
+CREATE INDEX `idx_contracts_archival_reason` ON `contracts_temp`(`archival_reason`);
+CREATE INDEX `idx_contracts_fcid` ON `contracts_temp`(`fcid`);
+CREATE INDEX `idx_contracts_host_key` ON `contracts_temp`(`host_key`);
+CREATE INDEX `idx_contracts_proof_height` ON `contracts_temp`(`proof_height`);
+CREATE INDEX `idx_contracts_renewed_from` ON `contracts_temp`(`renewed_from`);
+CREATE INDEX `idx_contracts_renewed_to` ON `contracts_temp`(`renewed_to`);
+CREATE INDEX `idx_contracts_revision_height` ON `contracts_temp`(`revision_height`);
+CREATE INDEX `idx_contracts_start_height` ON `contracts_temp`(`start_height`);
+CREATE INDEX `idx_contracts_state` ON `contracts_temp`(`state`);
+CREATE INDEX `idx_contracts_window_end` ON `contracts_temp`(`window_end`);
+CREATE INDEX `idx_contracts_window_start` ON `contracts_temp`(`window_start`);
+
+
+INSERT INTO contracts_temp (
+    id,
+    created_at,
+    fcid,
+    host_key,
+    proof_height,
+    renewed_from,
+    revision_height,
+    revision_number,
+    size,
+    start_height,
+    state,
+    window_start,
+    window_end,
+    contract_price,
+    initial_renter_funds,
+    delete_spending,
+    fund_account_spending,
+    sector_roots_spending,
+    upload_spending
+) SELECT
+    c.id,
+    c.created_at,
+    c.fcid,
+    h.public_key,
+    c.proof_height,
+    c.renewed_from,
+    c.revision_height,
+    c.revision_number,
+    c.size,
+    c.start_height,
+    c.state,
+    c.window_start,
+    c.window_end,
+    c.contract_price,
+    c.total_cost,
+    c.delete_spending,
+    c.fund_account_spending,
+    c.list_spending,
+    c.upload_spending
+FROM contracts c
+INNER JOIN hosts h ON c.host_id = h.id;
+
+INSERT INTO contracts_temp (
+    created_at,
+    archival_reason,
+    fcid,
+    host_key,
+    proof_height,
+    renewed_from,
+    renewed_to,
+    revision_height,
+    revision_number,
+    size,
+    start_height,
+    state,
+    window_start,
+    window_end,
+    contract_price,
+    initial_renter_funds,
+    delete_spending,
+    fund_account_spending,
+    sector_roots_spending,
+    upload_spending
+) SELECT
+    ac.created_at,
+    ac.reason,
+    ac.fcid,
+    COALESCE(h.public_key, X'0000000000000000000000000000000000000000000000000000000000000000'),
+    ac.proof_height,
+    ac.renewed_from,
+    ac.renewed_to,
+    ac.revision_height,
+    ac.revision_number,
+    ac.size,
+    ac.start_height,
+    ac.state,
+    ac.window_start,
+    ac.window_end,
+    ac.contract_price,
+    ac.total_cost,
+    ac.delete_spending,
+    ac.fund_account_spending,
+    ac.list_spending,
+    ac.upload_spending
+FROM `archived_contracts` ac
+LEFT JOIN hosts h ON ac.host = h.public_key;
+
+DROP TABLE `archived_contracts`;
+
+PRAGMA foreign_keys = OFF;
```

Comment:
Theoretically you shouldn't need this PRAGMA anymore since our SQLite migrations are now implicitly wrapped in it. Together with a `PRAGMA foreign_key_check`